server/setup_flow.go

```diff
@@ -0,0 +1,386 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-api/experimental/flow"
+	"github.com/mattermost/mattermost-server/v6/model"
+
+	"github.com/mattermost/mattermost-plugin-jira/server/utils/types"
+)
+
+const (
+	stepSetupWelcome             flow.Name = "setup-welcome"
+	stepDelegate                 flow.Name = "delegate"
+	stepDelegated                flow.Name = "delegated"
+	stepChooseEdition            flow.Name = "choose-edition"
+	stepCloudAddedInstance       flow.Name = "cloud-added"
+	stepCloudEnableDeveloperMode flow.Name = "cloud-enable-dev"
+	stepCloudUploadApp           flow.Name = "cloud-upload-app"
+	stepCloudInstalledApp        flow.Name = "cloud-installed"
+	stepCloudConnect             flow.Name = "cloud-connect"
+	stepCloudConnected           flow.Name = "cloud-connected"
+	stepConfigureServerApp       flow.Name = "configure-server-app"
+	stepWebhook                  flow.Name = "webhook"
+	stepWebhookDone              flow.Name = "webhook-done"
+	stepCancel                   flow.Name = "cancel"
+	stepDone                     flow.Name = "done"
+)
+
+const (
+	keyEdition             = "Edition"
+	keyDelegatedTo         = "Delegated"
+	keyJiraURL             = "URL"
+	keyConnectURL          = "ConnectURL"
+	keyWebhookURL          = "WebhookURL"
+	keyManageWebhooksURL   = "ManageWebhooksURL"
+	keyAtlassianConnectURL = "ACURL"
+)
+
+func (p *Plugin) NewSetupFlow() flow.Flow {
+	pluginURL := *p.client.Configuration.GetConfig().ServiceSettings.SiteURL + "/" + "plugins" + "/" + manifest.ID
+	conf := p.getConfig()
+	return flow.NewUserFlow("setup-wizard", p.client, pluginURL, conf.botUserID).
+		WithSteps(
+			p.stepWelcome(),
+			p.stepDelegate(),
+			p.stepDelegated(),
+			p.stepChooseEdition(),
+
+			// Jira Cloud steps
+			p.stepCloudAddedInstance(),
+			p.stepCloudEnableDeveloperMode(),
+			p.stepCloudUploadApp(),
+			p.stepCloudInstalledApp(),
+
+			p.stepCloudConnect(),
+			p.stepCloudConnected(),
+
+			p.stepWebhook(),
+			p.stepWebhookDone(),
+
+			p.stepCancel(),
+			p.stepDone(),
+		).
+		InitHTTP(p.gorillaRouter)
+}
+
+var cancelButton = flow.Button{
+	Name:    "Cancel",
+	Color:   flow.ColorDanger,
+	OnClick: flow.Goto(stepCancel),
+}
+
+func continueButtonF(f func(f flow.Flow) (flow.Name, flow.State)) flow.Button {
+	return flow.Button{
+		Name:    "Continue",
+		Color:   flow.ColorPrimary,
+		OnClick: f,
+	}
+}
+
+func continueButton(next flow.Name) flow.Button {
+	return continueButtonF(flow.Goto(next))
+}
+
+func skipButtonF(f func(f flow.Flow) (flow.Name, flow.State)) flow.Button {
+	return flow.Button{
+		Name:    "DEBUG Skip",
+		Color:   flow.ColorWarning,
+		OnClick: f,
+	}
+}
+
+func skipButton(next flow.Name) flow.Button {
+	return skipButtonF(flow.Goto(next))
+}
+
+func (p *Plugin) stepWelcome() flow.Step {
```

Comment:
@levb how do we see this in action?  Can we get a demo instance @DHaussermann ?

Reply:
see screenshots attached, otherwise need Jira admin etc.